### PR TITLE
148 - Add `run` method to PromptLayer class for executing prompts

### PR DESCRIPTION
**Pull request by abubakarsohail**
This pull request adds a new `run` method to the `PromptLayer` class. The `run` method allows for executing prompts with various parameters such as prompt name, version, label, input variables, tags, metadata, and group ID. It utilizes the existing `templates` and `track` modules to retrieve the prompt template and track the request. This new method enhances the functionality of the `PromptLayer` class and provides a convenient way to execute prompts.

### CODE REVIEW

File: promptlayer/utils.py [hunk 1]
```diff
@@ -642,3 +642,68 @@ def get_all_prompt_templates(
...
+MAP_TYPE_TO_OPENAI_FUNCTION = {
+    "chat": openai_chat_request,
+    "completion": openai_completions_request,
+}
+
+
+def openai_request(prompt_blueprint: GetPromptTemplateResponse, **kwargs):
+    from openai import OpenAI
+
+    client = OpenAI()
```
Comment (Jped) [inline promptlayer/utils.py:681]: @abubakarsohail if we do base_url this is where that information would go.

Comment (Jped) [reply promptlayer/utils.py:681]: thinking outloud here this gets complicated if the user has to authenticate their base_url

Comment (abubakarsohail) [reply promptlayer/utils.py:681]: I will look into where base url is available in prompt blueprint and plug that in. 
About authentication, as we discussed, we are not supporting every use case from the beginning and trying to keep it very simple.

Comment (abubakarsohail) [reply promptlayer/utils.py]: @Jped base url support is added now

**Review by Jped — APPROVED** [review]

**Pull request merged**